core/src/main/resources/defaults/blueprints/cdp-sdx-ha.bp

```diff
@@ -0,0 +1,433 @@
+            "base": true,
+            "refName": "kafka-GATEWAY-BASE",
+            "roleType": "GATEWAY"
+          },
+          {
+            "base": true,
+            "refName": "kafka-KAFKA_BROKER-BASE",
+            "roleType": "KAFKA_BROKER"
+          }
+        ],
+        "serviceConfigs": [
+          {
+            "name": "zookeeper_service",
+            "ref": "zookeeper"
+          },
+          {
+            "name": "offsets.topic.replication.factor",
+            "value": "1"
+          }
+        ],
+        "serviceType": "KAFKA"
+      },
+      {
+        "refName": "ranger",
+        "roleConfigGroups": [
+          {
+            "base": true,
+            "refName": "ranger-RANGER_USERSYNC-BASE",
+            "roleType": "RANGER_USERSYNC"
+          },
+          {
+            "base": true,
+            "refName": "ranger-RANGER_TAGSYNC-BASE",
+            "roleType": "RANGER_TAGSYNC"
+          },
+          {
+            "base": true,
+            "refName": "ranger-RANGER_ADMIN-BASE",
+            "roleType": "RANGER_ADMIN"
+          }
+        ],
+        "serviceConfigs": [
+          {
+            "name": "hdfs_service",
+            "ref": "hdfs"
+          },
+          {
+            "name": "rangeradmin_user_password",
+            "value": "{{{ general.password }}}"
+          },
+          {
+            "name": "rangertagsync_user_password",
+            "value": "{{{ general.password }}}"
+          },
+          {
+            "name": "solr_service",
+            "ref": "solr"
+          },
+          {
+            "name": "rangerusersync_user_password",
+            "value": "{{{ general.password }}}"
+          },
+          {
+            "name": "keyadmin_user_password",
+            "value": "{{{ general.password }}}"
+          }
+        ],
+        "serviceType": "RANGER"
+      },
+      {
+        "refName": "solr",
+        "roleConfigGroups": [
+          {
+            "refName": "solr-SOLR_SERVER-BASE",
+            "roleType": "SOLR_SERVER",
+            "base": true
+          }
+        ],
+        "serviceConfigs": [
+          {
+            "name": "hdfs_service",
+            "ref": "hdfs"
+          },
+          {
+            "name": "zookeeper_service",
+            "ref": "zookeeper"
+          }
+        ],
+        "serviceType": "SOLR"
+      },
+      {
+        "refName": "zookeeper",
+        "serviceType": "ZOOKEEPER",
+        "roleConfigGroups": [
+          {
+            "refName": "zookeeper-SERVER-BASE",
+            "roleType": "SERVER",
+            "base": true
+          }
+        ]
+      },
+      {
+        "serviceType": "KNOX",
+        "refName": "knox",
+        "roleConfigGroups": [
+          {
+            "base": true,
+            "refName": "knox-KNOX_GATEWAY-BASE",
+            "roleType": "KNOX_GATEWAY",
+            "configs": [
+              {
+                "name": "gateway_master_secret",
+                "value": "cloudera1"
```

Comment:
Is it intentional?

Reply:
@keyki may be able to correct me, but I think KNOX `gateway_master_secret` is automatically injected, no need to hard-code it in the template.

https://github.com/hortonworks/cloudbreak/blob/12265ad4729b4b48e75422497a9f9415d897fe7f/template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/knox/KnoxGatewayConfigProvider.java#L45

Reply:
When we added this part to the template it was not injected automatically so we had to add it in the template. I will try to remove it and check.

Reply:
Yeah, so there is a safety check in the code
```
if (gateway != null) {
```
then we configure the master secret. I think that single line that configures it should be moved out from the condition. Gateway is only not null when you enable it on CB API
```
    "gateway": {
      "topologies": [
        {
          "topologyName": "dp-proxy",
          "exposedServices": [
            "ALL"
          ]
        }
      ],
      "ssoType": "NONE"
    }
```
otherwise it will not set it now.

Reply:
@akanto @adoroszlai I removed the "gateway_master_secret" from the blueprint to see if it works.

Cluster creation failed with below error.
: Command failed to run because this role has an invalid configuration. Review and correct its configuration. First error: Missing required value: Knox Master Secret, Command [Start], with id [119] failed: Command failed to run because this role has an invalid configuration. Review and correct its configuration. First error: Missing required value: IDBroker Master Secret]

Reply:
OK, thanks for checking.

Reply:
@adoroszlai I will close this pull request and submit another one.